### PR TITLE
CRM-19490: Profile date fields don't respect localisation on the Cont…

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
…ribution Page confirmation screen

@seamuslee001 This is a narrow fix for the confirmation screen building on the stuff we've been working on .

The change to the Date file is just a tidy up of a previous commit - I was adding datepicker to all fields rather than only those that need them

The srtftime (posix?) format ALWAYS has a '%' in it - so we can actually make crmDate cope with either posix format or php date format quite easily, however some confusion has come through discussions about whether to use the format configured for the field for view purposes. Discussed with @seamuslee001 & he made the case that the shortDateFormat is most consistent for view purposes

### CODE REVIEW

File: templates/CRM/UF/Form/Fields.tpl [hunk 1]
```diff
@@ -93,12 +93,15 @@
           ( $form.formName neq 'Confirm' )  AND
           ( $form.formName neq 'ThankYou' ) }
             {include file="CRM/common/jcalendar.tpl" elementName=$profileFieldName}
-          {elseif ( $profileFieldName|substr:-5:5 eq '_date' ) }
+          {elseif $field.is_datetime_field && $action == 4}
```
Comment (eileenmcnaughton) [inline templates/CRM/UF/Form/Fields.tpl]: $action == 4 - ie we are looking at a page with the action view- should be the case for confirm screens

**Comment by seamuslee001** [comment]
@eileenmcnaughton in my testing this seems to revert the localisation on birth date field on confirm page

**Comment by seamuslee001** [comment]
Before before patch
After after patch

**Comment by eileenmcnaughton** [comment]
So for the date field - do you have any config in the database for it.

**Comment by eileenmcnaughton** [comment]
The fallback is onto this for formatting - if birth date is set to the default

screenshot 2017-02-24 11 53 36

**Comment by seamuslee001** [comment]
@eileenmcnaughton we do have that set up and the date picker correctly uses that in Main.php but then in the Confirm.php step we see it defaulting to American format in the display Even tho its not clear why from the settings. 

Attached are the date format / date dispaly settings on the test site i'm working on and the birth date preferences
date display settings
birth date preference setting

**Comment by seamuslee001** [comment]
@eileenmcnaughton also that falling back to default for the Confirm purpose doesn't make sense to me as we are displaying a date here not doing input for a date

**Comment by eileenmcnaughton** [comment]
@seamuslee001 so, I think the latest did what you said - ie. puts that formatting back into the tpl for view & makes it consistent with what has been done in the past. I'm still a bit confused that we are ignoring the custom format for custom fields - but I think you are saying that is consistent with core fields - which makes sense

File: templates/CRM/UF/Form/Fields.tpl [hunk 2]
```diff
@@ -89,16 +89,17 @@
             {include file="CRM/Profile/Form/GreetingType.tpl"}
           {elseif ($profileFieldName eq 'group' && $form.group) || ($profileFieldName eq 'tag' && $form.tag)}
             {include file="CRM/Contact/Form/Edit/TagsAndGroups.tpl" type=$profileFieldName title=null context="profile"}
-          {elseif ( $profileFieldName|substr:-5:5 eq '_date' ) AND
-          ( $form.formName neq 'Confirm' )  AND
-          ( $form.formName neq 'ThankYou' ) }
-            {include file="CRM/common/jcalendar.tpl" elementName=$profileFieldName}
-          {elseif ( $profileFieldName|substr:-5:5 eq '_date' ) }
+          {elseif $field.is_datetime_field && $action == 4}
```
Comment (seamuslee001) [inline templates/CRM/UF/Form/Fields.tpl]: @eileenmcnaughton for me this is a problem specifically $action == 4 that is what is stopping it from looking right on the confirm page. 

When print out what I think the action maybe it = 1

Comment (eileenmcnaughton) [inline templates/CRM/UF/Form/Fields.tpl]: It that is not being triggered on the Confirm page then you should be seeing a datepicker field on the Confirm page. the view-version of the field is only displayed if that bit of code is triggered

**Review by seamuslee001 — APPROVED** [review]
Revised changes look good to me and seem to work on my test site and there doesn't appear to be any obvious regression

**Comment by eileenmcnaughton** [comment]
The jenkins job is showing a success but it hasn't updated to github. I checked the jenkins job & I think the problem is just the bridge to github so merging

**Pull request merged**